pallets/author-slot-filter/src/lib.rs

```diff
@@ -41,6 +41,7 @@ mod mock;
 #[cfg(test)]
 mod tests;
 
+#[allow(deprecated)]
```

Comment:
```suggestion
```

This doesn't seem a good idea, because in the future deprecated items will be removed and we will have to fix the code anyway, so it is better to fix it now.

Reply:
Yes I agree. I put that `#[allow(deprecated)]` in the pallet because a warning is thrown in the definition of this storage item:

```rust
#[pallet::storage]
#[pallet::getter(fn eligible_ratio)]
#[deprecated(note = "use `pallet::EligibleCount` instead")]
pub type EligibleRatio<T: Config> = StorageValue<_, Percent, ValueQuery, Half<T>>;
```
I didn't find a way to remove the warning without using `#[allow(deprecated)]`. Is there any other way you suggest?

Reply:
The warning mentions using `pallet::EligibleCount` instead, did you try that?

Reply:
Yes, that is another storage item that is defined below in the code, so the solution is not to use `EligibleCount` instead, the warning is thrown in the definition of `EligibleRatio`. The actual solution would be to remove the storage item entirely but I guess this would cause problems if other teams are still using it.

Reply:
You shouldn't see a warning here, that's strange. Can you try removing this allow, or if that doesn't work moving the allow to the line that emits the warning?

Reply:
Ok it seems that this warning has been here for over a year, and it's probably caused by some macro. So let's remove the `allow(deprecated)` and keep the warning.